### PR TITLE
Issue #6916: migrate integration tests to junit5

### PR DESCRIPTION
**Pull request by pbludov**
issue #6916 

An attempt to fix `it` tests with junit5.

### CODE REVIEW

File: src/it/java/com/google/checkstyle/test/chapter2filebasic/rule21filename/OuterTypeFilenameTest.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 import static com.puppycrawl.tools.checkstyle.checks.OuterTypeFilenameCheck.MSG_KEY;
 
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/chapter2filebasic/rule21filename/OuterTypeFilenameTest.java]: If the old class is forbidden, I would add that to the import control to ban.

File: src/it/java/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/MethodTypeParameterNameTest.java [hunk 2]
```diff
@@ -21,8 +21,8 @@
 
 import java.util.Map;
 
-import org.junit.BeforeClass;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeAll;
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/MethodTypeParameterNameTest.java]: same as previous and other classes.

File: src/it/java/com/google/checkstyle/test/chapter2filebasic/rule21filename/OuterTypeFilenameTest.java (hunk 1 above)
Comment (pbludov) [reply src/it/java/com/google/checkstyle/test/chapter2filebasic/rule21filename/OuterTypeFilenameTest.java]: done

File: src/it/java/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/MethodTypeParameterNameTest.java (hunk 2 above)
Comment (pbludov) [reply src/it/java/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/MethodTypeParameterNameTest.java]: Done. Now we have two import control configs: one for `com.google/sun` another for `org.checkstyle`

**Comment by pbludov** [comment]
> Ahh, there is a same issue as I faced when I tried to move tests - it skips some tests for some reason.

This is due to missing jupiter engine for junit5. See
https://stackoverflow.com/questions/47079105/how-do-i-run-junit-5-integration-tests-with-the-maven-failsafe-plugin

File: src/it/java/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/MethodTypeParameterNameTest.java (hunk 2 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/MethodTypeParameterNameTest.java]: @pbludov I'm sorry, I didn't notice this was integration only changes. I don't think we need 2 import control configs.
If we are doing this in groups in separate PR (tests, int1, int2, etc..), we can postpone the import control or add a suppression for areas not done yet. @romani likes the suppression more so as it prevents others from undoing what you have already started.

Comment (pbludov) [reply src/it/java/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/MethodTypeParameterNameTest.java]: Done. Actually, once this issue will be done, all `junit4` packages will not be accessible at all as the reference to the module `junit4` will be removed from the `pom.xml` file.

**Pull request merged**